packages/plugin-polyfills/src/index.js

```diff
@@ -15,21 +21,41 @@ class PolyfillsResource extends ResourceInterface {
   async optimize(url, body) {
     return new Promise(async (resolve, reject) => {
       try {
-        const { projectDirectory, userWorkspace } = this.compilation.context;
-        const dependencies = fs.existsSync(path.join(userWorkspace, 'package.json')) // handle monorepos first
-          ? JSON.parse(fs.readFileSync(path.join(userWorkspace, 'package.json'), 'utf-8')).dependencies
-          : fs.existsSync(path.join(projectDirectory, 'package.json'))
-            ? JSON.parse(fs.readFileSync(path.join(projectDirectory, 'package.json'), 'utf-8')).dependencies
-            : {};
-        const litPolyfill = dependencies && dependencies.lit
-          ? '<script src="/node_modules/lit/polyfill-support.js"></script>\n'
-          : '';
+        let newHtml = body;
+
+        // standard WC polyfill
+        if (this.options.wc) {
+          newHtml = newHtml.replace('<head>', `
+            <head>
+              <script src="/node_modules/@webcomponents/webcomponentsjs/webcomponents-loader.js"></script>
+          `);
+        }
+
+        // append Lit polyfill next to make sure it comes before WC polyfill
+        if (this.options.lit) {
+          newHtml = newHtml.replace('<head>', `
+            <head>
+              <script src="/node_modules/lit/polyfill-support.js"></script>
+          `);
+        }
 
-        const newHtml = body.replace('<head>', `
-          <head>
-            ${litPolyfill}
-            <script src="/node_modules/@webcomponents/webcomponentsjs/webcomponents-loader.js"></script>
-        `);
+        // lastly, Declarative Shadow DOM polyfill
+        if (this.options.dsd) {
+          newHtml = newHtml.replace('</body>', `
```

Comment:
Hmm, this should probably test for ShadowRoot support first, and avoid if not.  Or maybe this comes from webcomponents package?